src/EnergyPlus/HVACUnitaryBypassVAV.cc

```diff
@@ -616,8 +616,8 @@ namespace HVACUnitaryBypassVAV {
                 thisCBVAV.FanVolFlow = 9999.0;
             } else {
                 if (thisCBVAV.FanType == DataHVACGlobals::FanType::System) {
-                    thisCBVAV.FanIndex = HVACFan::getFanObjectVectorIndex(state, thisCBVAV.FanName);
                     state.dataHVACFan->fanObjs.emplace_back(new HVACFan::FanSystem(state, thisCBVAV.FanName)); // call constructor
+                    thisCBVAV.FanIndex = HVACFan::getFanObjectVectorIndex(state, thisCBVAV.FanName);
                     thisCBVAV.FanInletNodeNum = state.dataHVACFan->fanObjs[thisCBVAV.FanIndex]->inletNodeNum;
                     fanOutletNode = state.dataHVACFan->fanObjs[thisCBVAV.FanIndex]->outletNodeNum;
                     thisCBVAV.FanVolFlow = state.dataHVACFan->fanObjs[thisCBVAV.FanIndex]->designAirVolFlowRate;
```

Comment:
Conversion of model getInput to eliminate mining functions uses getFanInex first for legacy fans and emplace_back THEN getFanIndex for Fan:SystemModel.

Reply:
Just saying, but `thisCBVAV.FanIndex = state.dataHVACFan->fanObjs.size() - 1;` should work and be less computationally intensive

Reply:
I was hoping to reduce all the mining functions to a single getIndex for each component type so all getInput code would look the same But your right, here, right after the emplace_back, the index is as you say, size - 1.

Reply:
Sure, this could do for a cleanup, but not today.  Thanks @rraustad